tests/backend/test_nongaussian_gates.py

```diff
@@ -18,13 +18,49 @@
 
 import numpy as np
 
-KAPPAS = np.linspace(0, 2 * np.pi, 7)
+from scipy.linalg import expm
 
+KAPPAS = np.linspace(0, 2 * np.pi, 7)
+GAMMAS = np.linspace(0, 2 * np.pi, 7)
 
 @pytest.mark.backends("fock", "tf")
 class TestFockRepresentation:
     """Tests that make use of the Fock basis representation."""
 
+    @pytest.mark.parametrize("gamma", GAMMAS)
+    def test_cubic_phase(self, setup_backend, gamma, cutoff, tol):
+        """Tests if the Cubic phase gate has the right effect on states in the Fock basis"""
+
+        backend = setup_backend(1)
+
+        backend.prepare_ket_state(np.ones([cutoff]) / np.sqrt(cutoff), 0)
+        backend.cubic_phase(gamma, 0)
+        s = backend.state()
+        if s.is_pure:
+            numer_state = s.ket()
+        else:
+            numer_state = s.dm()
+
+        #Create annihilation operator matrix
+        ladder_vals = np.arange(1, cutoff)
+        ladder_vals = np.sqrt(ladder_vals)
+        a = np.zeros([cutoff, cutoff])
+        np.fill_diagonal(a[:, 1:], ladder_vals)
+
+        #Construct (unnormalized) x matrix, ie a+a^dag, and it's third power
+        x = a + np.transpose(a)
+        x3 = np.matmul(x, np.matmul(x, x))
```

Comment:
Note that this can equivalently be written
```python
x = a + a.T
x3 = x @ x @ x
```
Either style is fine, but we tend to err on the side of readability :slightly_smiling_face:

Reply:
Cool, thanks. I'll use that next time :)